controllers/bpfman-agent/common.go

```diff
@@ -519,32 +528,47 @@ func (r *ReconcilerCommon) getExistingBpfPrograms(ctx context.Context,
 		return nil, err
 	}
 
-	existingBpfPrograms := map[string]bpfmaniov1alpha1.BpfProgram{}
+	existingBpfPrograms := map[bpfProgKey]bpfmaniov1alpha1.BpfProgram{}
 	for _, bpfProg := range bpfProgramList.Items {
-		existingBpfPrograms[bpfProg.GetName()] = bpfProg
+		key := bpfProgKey{
+			appProgId:   bpfProg.GetLabels()[internal.AppProgramId],
+			attachPoint: bpfProg.GetAnnotations()[internal.BpfProgramAttachPoint],
```

Comment:
just curious why u need both progId and attachement point as key, progId should be unique right ?

Reply:
Not exactly.  I'm trying to use appProgId to uniquely identify a single program item within an ApplicationProgram.  For this, I'm currently using program type plus all or a portion of the attach point.  In some cases (Fentry, Fexit, Kprobe) it's the whole attach point.  In the case of uprobes, it's the FunctionName (I think I should also add the target), but it can actually be attached to the same point in multiple containers (each one results in a separate BpfProgram object).  In the case of tracepoint, it's the first tracepoint on on the list.  In the case of XDP, it's the first interface on the list.  In the case of TC, it's the first interface on the list plus the direction.  This kind of works because we've said that there will only ever be one program of a given type per attach point.  However, that may not hold true for everyone, and for the program types where there is a list of attach points, I wonder if we'd ever have two programs of the same type that have different lists, but the first item is the same. That probably won't happen with our applications, but I don't know if we can rule it out.  My assumption is that we'll need to change something in the future to make this 100% unique.  One Idea I have is to have the operator controller generate a unique ID for each program.  Another is to require that the user providd a unique name/id for each program in a BpfApplication.

Reply:
So, to summarize, the appProgId should uniquely identify a program item in a BpfApplication, and the following set of info uniquely identifies a BpfProgram object that was created for it:

Labels:
- internal.BpfProgramOwner
- internal.AppProgramId
- internal.K8sHostLabel

Annotation:
- internal.BpfProgramAttachPoint

I put the attach point in an annotation because we don't need to use it in a search, and it can be a longer string if needed.